core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -938,6 +944,19 @@ trait EntityComponent {
         entityAction map { _.toSet map { e: Entity => e.toReference } }
       }
     }
+
+    def countReferringEntities(context: Workspace, entityIds: Set[Long]): ReadAction[Long] = {
```

Comment:
My original implementation for this endpoint piggy-backed off of what the existing deleteEntities endpoint was doing. However, I realized quickly that it didn't perform well. When I looked into why- it was doing recursive queries to get the entire entity tree for each of the entities being deleted. I don't think that this is necessary, so I short-circuited it and am only checking to see if there are any direct references to the entities being deleted